cmdget.go

```diff
@@ -330,3 +345,81 @@ func removeAutoGoMod(m *moduleVCSInfo) (bool, error) {
 func autoGoMod(mpath string) string {
 	return "// Generated by gohack; DO NOT EDIT.\nmodule " + mpath + "\n"
 }
+
+// checkCanReplace checks whether it may be possible to replace
+// the module in the given go.mod file.
+func checkCanReplace(f *modfile.File, mod string) error {
+	var found *modfile.Replace
+	for _, r := range f.Replace {
+		if r.Old.Path != mod {
+			continue
+		}
+		if found != nil {
+			return errors.Newf("found multiple existing replacements for %q", mod)
+		}
+		if r.New.Version == "" {
+			return errors.Newf("%s is already replaced by %s", mod, r.New.Path)
+		}
+	}
+	return nil
+}
+
+// replaceModule adds or modifies a replace statement in f for mod
+// to be replaced with dir.
+func replaceModule(f *modfile.File, mod string, dir string) error {
+	var found *modfile.Replace
+	for _, r := range f.Replace {
+		if r.Old.Path != mod {
+			continue
+		}
+		// These checks shouldn't fail when checkCanReplace has been
+		// called previously, but check anyway just to be sure.
```

Comment:
then maybe make them panics? otherwise you may have a bug in checkCanReplace, and you might never notice.

Or perhaps make these error messages more scary, like "this shouldn't happen, file a bug!". Though I tend to just make these be panics.

Reply:
done